packages/editor/src/components/rich-text/format-toolbar/index.js

```diff
@@ -22,176 +11,49 @@ import { FORMATTING_CONTROLS } from '../formatting-controls';
-	}
-
-	setLinkTarget( opensInNewWindow ) {
-		this.setState( { opensInNewWindow } );
-		if ( this.props.formats.link && ! this.props.formats.link.isAdding ) {
-			this.props.onChange( { link: {
-				value: this.props.formats.link.value,
-				target: opensInNewWindow ? '_blank' : null,
-				rel: opensInNewWindow ? 'noreferrer noopener' : null,
-			} } );
-		}
-	}
-
-	addLink() {
-		this.props.onChange( { link: { isAdding: true } } );
-	}
-
-	dropLink() {
-		this.props.onChange( { link: null } );
-	}
-
-	editLink( event ) {
-		event.preventDefault();
-		this.setState( { linkValue: this.props.formats.link.value, isEditingLink: true } );
-	}
-
-	submitLink( event ) {
-		event.preventDefault();
-		const value = prependHTTP( this.state.linkValue );
-		this.props.onChange( { link: {
-			isAdding: false,
-			target: this.state.opensInNewWindow ? '_blank' : null,
-			rel: this.state.opensInNewWindow ? 'noreferrer noopener' : null,
-			value,
-		} } );
-		if ( ! this.props.formats.link.value ) {
-			this.props.speak( __( 'Link added.' ), 'assertive' );
-		}
-	}
-
-	isFormatActive( format ) {
-		return this.props.formats[ format ] && this.props.formats[ format ].isActive;
-	}
-
-	render() {
-		const { formats, enabledControls = [], customControls = [], selectedNodeId } = this.props;
-		const { linkValue, settingsVisible, opensInNewWindow, isEditingLink } = this.state;
-		const isAddingLink = formats.link && formats.link.isAdding;
-		const isEditing = isAddingLink || isEditingLink;
-		const isPreviewing = ! isEditing && formats.link;
-
-		const toolbarControls = FORMATTING_CONTROLS.concat( customControls )
-			.filter( ( control ) => enabledControls.indexOf( control.format ) !== -1 )
-			.map( ( control ) => {
-				if ( control.format === 'link' ) {
-					const isFormatActive = this.isFormatActive( 'link' );
-					const isActive = isFormatActive || isAddingLink;
-					return {
-						...control,
-						icon: isFormatActive ? 'editor-unlink' : 'admin-links', // TODO: Need proper unlink icon
-						title: isFormatActive ? __( 'Unlink' ) : __( 'Link' ),
-						onClick: isActive ? this.dropLink : this.addLink,
-						isActive,
-					};
-				}
-
+const FormatToolbar = ( props ) => {
+	const link = props.getActiveFormat( 'a' );
+	const removeLink = () => props.removeFormat( 'a' );
+	const addLink = () => props.applyFormat( {
+		type: 'a',
+		attributes: {
+			href: '',
+		},
+	} );
+
+	const toolbarControls = FORMATTING_CONTROLS
+		.filter( ( control ) => props.enabledControls.indexOf( control.format ) !== -1 )
+		.map( ( control ) => {
+			if ( control.format === 'link' ) {
 				return {
 					...control,
-					onClick: this.toggleFormat( control.format ),
-					isActive: this.isFormatActive( control.format ),
+					icon: link ? 'editor-unlink' : 'admin-links', // TODO: Need proper unlink icon
+					title: link ? __( 'Unlink' ) : __( 'Link' ),
+					onClick: link ? removeLink : addLink,
+					isActive: !! link,
```

Comment:
Mostly because `isActive` should be boolean, and `getActiveFormat` returns either an object with format info or `undefined`. Maybe we can check for that rather than doing `!!`.